pkg/controllers/replica-controller/handler.go

```diff
@@ -257,6 +261,24 @@ func (c *CStorVolumeReplicaController) cVREventHandler(
 		if isCVRCreateStatus(cvrObj) {
 			return c.cVRAddEventHandler(cvrObj, fullVolName)
 		}
+
+		if cvrObj.Annotations[restoreCompletedAnnotation] == "true" {
+			targetIp, err := volumereplica.GetTargetIp(fullVolName)
+			if err != nil {
+				return "", err
+			}
+
+			if targetIp == "" {
+				err := volumereplica.SetTargetIp(fullVolName, cvrObj.Spec.TargetIP)
+				if err != nil {
+					return "", err
+				}
+			}
+
+			delete(cvrObj.Annotations, restoreCompletedAnnotation)
+			delete(cvrObj.Annotations, volumereplica.IsRestoreVol)
```

Comment:
I am not sure if these changes in cvrObj are persisted to the etcd. I cannot see the Update/Patch workflow after the annotation delete. cc @mittachaitu

Reply:
Yeah, the caller is doing [that](https://github.com/openebs/cstor-operators/blob/f9e3ad47b6e0b4e86cbe1e1369f5384ee706df8e/pkg/controllers/replica-controller/handler.go#L134) work.